only show status when AC is offline

mgor opened the pull request:
Hey,

I wanted the possibility to only show battery status when the AC is offline. I made it configurable with `@batt_toggle_ac_status`, if not set it will be set to false and it would always show the battery status (how it works now).

This is done with the helper method `ac_online`, if it returns true (0) all the prefixes will return an empty string (not showing anything in the status bar), if it returns false (1) it will show the information.

I only had possibility to test on Ubuntu (with upower and checking sysfs). So `ac_online` should be updated with the results of using `pmset` or `acpi` as well.

Also, I fixed a bunch of shellcheck warnings.

Thanks for a great plugin!

martinbeentjes commented on scripts/helpers.sh:
```diff
@@ -33,9 +33,23 @@ battery_status() {
+    if command_exists "upower"; then
+        [[ "$(upower -i /org/freedesktop/UPower/devices/line_power_AC | awk -F: '/online/ {gsub(/ /, "", $NF); print $NF}')" == "yes" ]]
+    elif [[ -e /sys/class/power_supply/AC/online ]]; then
+        [[ "$(cat /sys/class/power_supply/AC/online)" -eq 1 ]]
```
What is the reason to use `-eq` here, while you changed those to `==` in the code above?

martinbeentjes commented on scripts/battery_status_bg.sh:
```diff
@@ -23,19 +23,21 @@ get_charge_color_settings() {
-    elif [ $percentage -le 50 -a $percentage -ge 16 ];then
-        printf $color_medium_charge
+    percentage=$("$CURRENT_DIR/battery_percentage.sh" | sed -e 's/%//')
+    if (( percentage == 100 )); then
```
Why do you choose to change the `-le` and such to the `<`?

martinbeentjes commented on scripts/battery_percentage.sh:
```diff
@@ -23,6 +25,6 @@ print_battery_percentage() {
 
 main() {
-	print_battery_percentage
+    print_battery_percentage
```
I should make this more clear, but we use tabs for indentation instead of four spaces.

mgor replied:
Ops, my bad! I'll have a look at that.

martinbeentjes commented on battery.tmux:
```diff
@@ -28,8 +28,9 @@ set_tmux_option() {
 	local all_interpolated="$1"
 	for ((i=0; i<${#battery_commands[@]}; i++)); do
-		all_interpolated=${all_interpolated/${battery_interpolation[$i]}/${battery_commands[$i]}}
+        all_interpolated=${all_interpolated/${battery_interpolation[$i]}/${battery_commands[$i]}}
```
We use tabs for indentation, not spaces.

martinbeentjes commented:
In your PR you are forcing all the *modules* of the plugin to automatically check something. I don't like this approach. I think it is better to leave the choice of actually using this new module to the user.

Also, is `/org/freedesktop/UPower/devices/line_power_AC` a general way of checking that AC information? Or is that specific to the distribution you are using?

mgor commented:
Do you have a suggestion on how to accomplish this? In a first version of the PR I did the check in `battery.tmux`, and replace every prefix with a empty string (instead of the helper script), but then I relalized that the values will never be updated by tmux, since that part is only run once (at start). Scripts specified with `#(...)` will run every update interval.

I'm not sure about `/org/freedesktop/UPower/devices/line_power_AC`, I made the assumption that it would be the same on every distribution with `upower` and `dbus`(?) :-)

martinbeentjes commented:
That statusbar of tmux is updated every five seconds (default interval), so that should also rerender all the information within the statusbar. 

![Statusbar](https://cloud.githubusercontent.com/assets/8865313/20647260/0b439ada-b490-11e6-915d-cedc60047845.gif)

mgor commented:
Hm... I think there might be some misunderstanding of this PR does? Or how do you suggest that one helper script would change the output for the other helper scripts?

Consider the status bar being:
`#{battery_status_bg} #{battery_percentage} #{battery_icon}`

If `ac_online` returns true (0), I don't want any output of either of these prefixes, and in `battery.tmux` these prefixes will be replaced by their respective helper script (hence, the check being made in each helper script).

If `ac_online` returns false (1), I want them to show the information.

martinbeentjes commented:
So the basic goal of this helper script is to check if the machine is connected to AC, and if so: not show any information?

mgor commented:
Yes, correct. But only if `@batt_toggle_ac_status` is set to `true`. In other words, if this configuration parameter isn't set, or if it is set to `false` it will behave as before.

martinbeentjes commented:
First of all, I don't really like the naming of the option. It does not indicate to me what it exactly does.  That is something that has to be changed before a possible merge.

Besides that, for now the PR adds the following: "Only show the battery information when the power adapter is **not** attached and `@batt_toggle_ac_status` to `true`." Can you confirm this?

mgor commented:
What's your suggestion on naming the option? I wanted to keep it as short as possible. 
batt_hide_when_ac_online?

Yes, I confirm that this is the case.

...and this is why it was added as an option, and doesn't change the default behaviour.

mgor commented:
I hope that I've fixed the issues you've pointed out. However, it seems like there was already some mixed-indent in `scripts/battery_status_bg.sh` (which was fixed now).

martinbeentjes commented:
> What's your suggestion on naming the option? I wanted to keep it as short as possible.
batt_hide_when_ac_online?

Maybe `batt_toggle_plug_status` is a bit more clear?

> Yes, I confirm that this is the case.

Okay, good!

> ...and this is why it was added as an option, and doesn't change the default behaviour.

Yes, now it makes more sense.

mgor commented:
>  Is this functionality present in upower?

I'm guessing you mean `acpi`? I installed `acpi` and added support for it.

Got SSH access to a friends Apple laptop and also added support for `pmset`.

martinbeentjes commented:
Well, that was quicker than I expected!

martinbeentjes commented on scripts/helpers.sh:
```diff
@@ -33,9 +33,26 @@ battery_status() {
+	elif command_exists "acpi"; then
+		acpi -a | grep -q 'on-line'
+	elif command_exists "pmset"; then
+		[[ $(pmset -g | awk '/^AC Power/ {gsub(/[^*]/, "", $NF); print $NF}') == '*' ]]
```
This should be `pmset -g batt`

mgor replied:
`pmset -g` reported both `AC Power` and `Batt Power` (among other things). What I could see was that a star (`*`) on the end of the line indicated if it was running on AC or battery. This was a 5 minute test.

martinbeentjes replied:
Hmm, but then still: check on the existence of `AC Power` would be sufficient to fix this right?

mgor replied:
Not sure which version of Mac OS X that was on the laptop I tried on, but the output was as follows:

    MacBook-Pro:~ mgor$ pmset -g | awk '/Power/'
    Battery Power           -1
    AC Power                -1*

Which is why I didn't have `pmset -g batt`.

But with your output, I would suggest to chain that `elif`-case to:

    pmset -g batt | egrep -q "^Now draining from.*AC Power"

(I hope that `egrep` is available in Mac OS X)

mgor commented:
Not sure which version of Mac OS X that was on the laptop I tried on, but the output was as follows:

    MacBook-Pro:~ mgor$ pmset -g | awk '/Power/'
    Battery Power           -1
    AC Power                -1*

Which is why I didn't have `pmset -g batt`.

But with your output, I would suggest to chain that `elif`-case to:

    pmset -g batt | egrep -q "^Now draining from.*AC Power"

(I hope that `egrep` is available in Mac OS X)

mgor commented:
Not sure which version of Mac OS X that was on the laptop I tried on, but the output was as follows:

    MacBook-Pro:~ mgor$ pmset -g | awk '/Power/'
    Battery Power           -1
    AC Power                -1*

Which is why I didn't have `pmset -g batt`.

But with your output, I would suggest to chain that `elif`-case to:

    pmset -g batt | egrep -q "^Now draining from.*AC Power"

(I hope that `egrep` is available in Mac OS X)

martinbeentjes commented:
I think you tested it on an earlier version than Sierra. `pmset` has been updated, the output of the command is different on macOS Sierra compared to OS X El Capitan.
